StackExchange.Redis/StackExchange/Redis/Interfaces/IDatabase.cs

```diff
@@ -1389,6 +1389,225 @@ public interface IDatabase : IRedis, IDatabaseAsync
+
+        /// <summary>
+        /// Retrieve information about the given stream. This is the equivalent of calling "XINFO STREAM key".
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>A <see cref="Redis.StreamInfo"/> instance with information about the stream.</returns>
+        /// <remarks>https://redis.io/topics/streams-intro</remarks>
+        StreamInfo StreamInfo(RedisKey key, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Retrieve information about the groups created for the given stream. This is the equivalent of calling "XINFO GROUPS key".
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>An instance of <see cref="Redis.StreamGroupInfo"/> for each of the stream's groups.</returns>
+        /// <remarks>https://redis.io/topics/streams-intro</remarks>
+        StreamGroupInfo[] StreamGroupInfo(RedisKey key, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Retrieve information about the consumers for the given consumer group. This is the equivalent of calling "XINFO GROUPS key group".
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="groupName">The consumer group name.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>An instance of <see cref="Redis.StreamConsumerInfo"/> for each of the consumer group's consumers.</returns>
+        /// <remarks>https://redis.io/topics/streams-intro</remarks>
+        StreamConsumerInfo[] StreamConsumerInfo(RedisKey key, RedisValue groupName, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Return the number of entries in a stream.
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>The number of entries inside the given stream.</returns>
+        /// <remarks>https://redis.io/commands/xlen</remarks>
+        long StreamLength(RedisKey key, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Delete messages in the stream.
+        /// </summary>
+        /// <param name="key">The key of the stream.</param>
+        /// <param name="messageIds">The IDs of the messages to delete.</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>Returns the number of messages successfully deleted from the stream.</returns>
+        /// <remarks>https://redis.io/topics/streams-intro</remarks>
+        long StreamMessagesDelete(RedisKey key, RedisValue[] messageIds, CommandFlags flags = CommandFlags.None);
```

Comment:
I think we can call this `StreamDelete`, as the `messageIds` are required and make it clear it's not deleting a stream.